contracts/voting/Voting.sol

```diff
@@ -0,0 +1,350 @@
+    // regionId => [city id]
+    mapping(uint256 => uint256[]) internal _regionToCities;
+
+
+    // TODO: change bool to the timestamp for calculating Monument reward?
+    // owner address => city id => Building => 0: not exists, >0: season number +1
+    mapping(address => mapping(uint256 => mapping(Building => uint256))) internal _ownerToBuildings;
+
+    // owner address => city id => season => claimed
+    mapping(address => mapping(uint256 => mapping(uint256 => bool))) internal _ownerClaims;
+
+    // city id => season id => Nominee[]
+    mapping(uint256 => mapping(uint256 => Nominee[])) internal _cityToNominees;
+
+    uint256 internal _cityIdCounter = 0;
+
+    constructor(
+        NFT mayorNFT,
+        Vote voteToken_,
+        Voucher voucherToken_,
+        uint256 votesPerCitizen_,
+        address owner
+    ) {
+        _mayor = mayorNFT;
+        _voteToken = voteToken_;
+        _voucherToken = voucherToken_;
+        _votesPerCitizen = votesPerCitizen_;
+        _owner = owner;
+        _voteDigits = _voteToken.decimals();
+    }
+
+    function transferTokens(address recipient) external override isOwner {
+        uint256 voteBalance = _voteToken.balanceOf(address(this));
+        uint256 voucherBalance = _voucherToken.balanceOf(address(this));
+        _voteToken.transfer(recipient, voteBalance);
+        _voucherToken.transfer(recipient, voucherBalance);
+    }
+
+    function changeVotesPerCitizen(uint256 amount) external override isOwner {
+        if (amount == 0) revert VotingErrors.IncorrectValue();
+        uint256 oldAmount = _votesPerCitizen;
+        _votesPerCitizen = amount;
+
+        emit VotesPerCitizenUpdated(oldAmount, amount);
+    }
+
+    function addCities(uint256 regionId, NewCity[] calldata newCities) external override isOwner {
+        if(newCities.length <= 0) revert VotingErrors.EmptyArray();
+        uint256[] memory cityIds = new uint256[](newCities.length);
+
+        for (uint256 i = 0; i < newCities.length; i++) {
+            NewCity memory newCity = newCities[i];
+            if (newCity.population == 0) revert VotingErrors.IncorrectValue();
+            City memory city = City({
+                regionId: regionId,
+                name: newCity.name,
+                population: newCity.population,
+                votePrice: newCity.votePrice,
+                active: true
+            });
+            _cities[_cityIdCounter] = city;
+            _regionToCities[regionId].push(_cityIdCounter);
+            cityIds[i] = _cityIdCounter;
+            _cityIdCounter++;
+        }
+        emit CitiesAdded(regionId, cityIds);
+        _startVoting(regionId);
+    }
+
+    function changeCityVotePrice(uint256 cityId, uint256 newPrice) external override isOwner {
+        City storage city = _cities[cityId];
+        if (city.population == 0) revert VotingErrors.UnknownCity();
+        if (_isVotingPeriod(city.regionId)) revert VotingErrors.IncorrectPeriod();
```

Comment:
Also these types of verifications better to move to separate funcs

Reply:
_isVotingPeriod() is already a separate function, or i don't understand what you mean